elastic/apm-server
# idxmgmt: fix ILM support detection

## Motivation/summary

If Elasticsearch is unavailable while checking for ILM support, we must not swallow errors and take them to mean ILM is not supported. This would lead to APM Server writing time-based indices for the remainder of an established connection.

Simplify idxgmmt package, and bring it closer into line with how the libbeat version works. Move ILM support checks into `Manager.Setup`, which will now return ILM check errors.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

This is a bit hard to test directly, as it depends on an error occurring with Elasticsearch at just the right time when calling the `/_xpack` endpoint to check that ILM is supported and enabled in Elasticsearch.

Perhaps: place a reverse proxy between apm-server and Elasticsearch which intermittently returns an error (not 400! this goes down a different code path) for `/_xpack`, and continuously restart apm-server and send some events, making sure no time-based indices are ever created.

## Related issues

Fixes #5259
